contents/data/sql/SQL.md

```diff
@@ -0,0 +1,145 @@
+
+**Structured Query Language** (**SQL**) is a programming language that is specifically designed for managing (storing, retrieving and manipulating) data in a <tooltip content="relational databases store data in tables">[relational database]({{baseUrl}}/contents/data/databases/databases.html#database-models)</tooltip>. Unlike other languages, SQL doesn't come as a standalone installation. Rather, _SQL Software_ (or _SQL Implementations_) like MySQL, Oracle, SQL Server and PostgreSQL come with a RDBMS that uses the SQL language.
+However, the syntaxes and functions used across these software systems <tooltip content="in terms of whether the syntax is case-sensitive, the format for specifying date and time, what functions are available out-of-the-box">vary</tooltip>.
+
+<box type="tip">
+  Many online resources refer to Microsoft SQL Server as SQL. So, keep that in mind before you start making conclusions about the characteristics and features of SQL, the language.
+</box>
+
+---
+
+### How does SQL work?
+
+SQL uses **queries** to retrieve data. Here is an example of how an SQL query is used.
+
+Suppose we have the data table `Students` shown here:
+
+| ID | Name  | Course | Faculty |
+| -- | ----- | ------ | ------- |
+| 1  | Alex  | CS202  | CS      |
+| 2  | Bob   | MA303  | MA      |
+| 3  | Cathy | CS202  | CS      |
+| 4  | Daren | CS202  | CS      |
+| 5  | Ellie | CS101  | CS      |
+| 6  | Fred  | MA303  | MA      |
+| 7  | Gary  | CS101  | CS      |
+| 8  | Henry | CS404  | CS      |
+
+We can use this SQL query to retrieve information from this table:
+```
+SELECT Course, COUNT(*) num
+FROM Students
+WHERE Faculty = 'CS'
+GROUP BY Course
+HAVING COUNT(*) > 1
+ORDER BY num;
+```
+
+<box type="tip">
+  The uppercase words in this query are either **clauses** or **statements**, and are SQL keywords.
+</box>
+
+This query first filters the entries in the `Students` table such that only entries that have `CS` as the faculty are considered.
+After that, it groups those entries into the 3 courses: `CS101`, `CS202` and `CS404`.
+Then, it removes courses that do not have more than 1 student, i.e., `CS404` is removed from consideration.
+Finally, it returns a list of courses with a count of the number of students, ordered in increasing order.
+So, the output of the query is:
+
+| Course | num |
+| ------ | --- |
+| CS202  | 3   |
+| CS101  | 2   |
+
+<box type="info">
+  You can experiment with this example on <a href="https://www.db-fiddle.com/f/kHqV2edUGxCc1dU6vE6CmS/0">DB Fiddle</a> by entering SQL queries and then running them.
+</box>
+
+You can see how this simple query can prove to be extremely useful for getting this information when the table has tens of entries (or many more). Apart from retrieving information, SQL can also be used for creating, deleting and manipulating data with queries like  `INSERT`, `DELETE` and `UPDATE` for entries, in addition to `CREATE`, `DROP` and `ALTER` for tables as a whole.
+
```

Comment:
This section is generally good. Perhaps give two more SQL statements of different types, with a one sentence explanation of each?